src/Raven.Server/Documents/Indexes/Persistence/Lucene/Documents/LuceneDocumentConverterBase.cs

```diff
@@ -191,7 +193,21 @@ protected IEnumerable<AbstractField> GetRegularFields(IndexField field, object v
         private IEnumerable<AbstractField> GetComplexObjectFields(string path, Field.Store storage, BlittableJsonReaderObject val, Field.Index indexing, Field.TermVector termVector)
         {
             yield return GetOrCreateField(path + ConvertToJsonSuffix, TrueString, null, null, storage, Field.Index.NOT_ANALYZED_NO_NORMS, Field.TermVector.NO);
-            yield return GetOrCreateField(path, null, null, val.GetJsonReader(), storage, indexing, termVector);
+
+            StreamReader inMemoryComplexObject;
+            if (_complexInMemoryObjects.TryGetValue(path, out inMemoryComplexObject) == false)
+            {
+                _complexInMemoryObjects[path] = inMemoryComplexObject = new StreamReader(new MemoryStream(), Encoding.UTF8, true, 1024, leaveOpen: true);  
+            }
+
+            var ms = inMemoryComplexObject.BaseStream;
+
+            ms.Position = 0;
+            val.WriteJsonTo(ms);
+            ms.SetLength(ms.Position);
+            ms.Position = 0;            
+
```

Comment:
The buffer is used, the buffered data is not.
This is what I meant:
http://referencesource.microsoft.com/#mscorlib/system/io/streamreader.cs,316

See the impl, it means that it will forget that it has data from the previous run